app/controllers/CalculationController.scala

```diff
@@ -110,20 +110,43 @@ class CalculationController @Inject()(cc: MessagesControllerComponents,
     }
   }
 
+  def createOverwriteCalculationListUrl(nino: String, taxYear: TaxYear): String = {
+    if (taxYear.endYear >= 2024) {
+      Logger("application").info(s"[CalculationController][createOverwriteCalculationListUrl] Overwriting calculation details TYS")
+      s"/income-tax/view/calculations/liability/${taxYear.formattedTaxYearRange}/$nino"
+    } else {
+      Logger("application").info(s"[CalculationController][createOverwriteCalculationListUrl] Overwriting calculation details legacy")
+      s"/income-tax/list-of-calculation-results/$nino?taxYear=${taxYear.endYearString}"
+    }
+  }
 
-  def overwriteCalculationListTYS(nino: String, taxYearRange: String, crystallisationStatus: String): Action[AnyContent] = Action.async { _ =>
+  def overwriteCalculationList(nino: String, taxYearRange: String, crystallisationStatus: String): Action[AnyContent] = Action.async { _ =>
```

Comment:
we need to refactor this method to make it shorter, ie. expression under dataRepository.replaceOne ..
can be moved into private function;